test/libsolutil/JSON.cpp

```diff
@@ -135,179 +135,202 @@ BOOST_AUTO_TEST_CASE(parse_json_strict)
 	// Trailing garbage is not allowed in ECMA-262
 	BOOST_CHECK(!jsonParseStrict("{\"1\":2,\"2\":\"2\",\"3\":{\"3.1\":\"3.1\",\"3.2\":3}}}}}}}}}}", json, &errors));
 
-	// Comments are not allowed in ECMA-262
-	// ... but JSONCPP allows them
-	BOOST_CHECK(jsonParseStrict(
-		"{\"1\":3, // awesome comment\n\"2\":\"2\",\"3\":{\"3.1\":\"3.1\",\"3.2\":5}}", json, &errors
-	));
-	BOOST_CHECK(json["1"] == 3);
-	BOOST_CHECK(json["2"] == "2");
-	BOOST_CHECK(json["3"]["3.1"] == "3.1");
-	BOOST_CHECK(json["3"]["3.2"] == 5);
-
 	// According to ECMA-404 object, array, number, string, true, false, null are allowed
 	// ... but JSONCPP disallows value types
 	BOOST_CHECK(jsonParseStrict("[]", json, &errors));
-	BOOST_CHECK(json.isArray());
+	BOOST_CHECK(json.is_array());
 	BOOST_CHECK(jsonParseStrict("{}", json, &errors));
-	BOOST_CHECK(json.isObject());
-	BOOST_CHECK(!jsonParseStrict("1", json, &errors));
-	// BOOST_CHECK(json.isNumeric());
-	BOOST_CHECK(!jsonParseStrict("\"hello\"", json, &errors));
-	// BOOST_CHECK(json.isString());
-	BOOST_CHECK(!jsonParseStrict("true", json, &errors));
-	// BOOST_CHECK(json.isBool());
-	BOOST_CHECK(!jsonParseStrict("null", json, &errors));
-	// BOOST_CHECK(json.isNull());
+	BOOST_CHECK(json.is_object());
+	BOOST_CHECK(jsonParseStrict("1", json, &errors));
+	BOOST_CHECK(json.is_number());
+	BOOST_CHECK(jsonParseStrict("\"hello\"", json, &errors));
+	BOOST_CHECK(json.is_string());
+	BOOST_CHECK(jsonParseStrict("true", json, &errors));
+	BOOST_CHECK(json.is_boolean());
+	BOOST_CHECK(jsonParseStrict("null", json, &errors));
+	BOOST_CHECK(json.is_null());
 
 	// Single quotes are also disallowed by ECMA-404
 	BOOST_CHECK(!jsonParseStrict("'hello'", json, &errors));
-	// BOOST_CHECK(json.isString());
+	// BOOST_CHECK(json.is_string());
 
 	// Only string keys in objects are allowed in ECMA-404
 	BOOST_CHECK(!jsonParseStrict("{ 42: \"hello\" }", json, &errors));
 
 	// According to ECMA-404 hex escape sequences are not allowed, only unicode (\uNNNN) and
 	// a few control characters (\b, \f, \n, \r, \t)
-	//
-	// More lenient parsers allow hex escapes as long as they translate to a valid UTF-8 encoding.
-	//
-	// ... but JSONCPP allows any hex escapes
```

Comment:
Ok, in each of these cases we should specifically confirm that we don't actually run into something that we'd really need to consider breaking.